dlt/pipeline/platform.py

```diff
@@ -0,0 +1,101 @@
+"""Implements SupportsTracking"""
+from typing import Any, cast, TypedDict, List
+import requests
+from dlt.common.managed_thread_pool import ManagedThreadPool
+from urllib.parse import urljoin
+
+from dlt.pipeline.trace import PipelineTrace, PipelineStepTrace, TPipelineStep, SupportsPipeline
+from dlt.common import json
+from dlt.common.runtime import logger
+from dlt.common.pipeline import LoadInfo
+from dlt.common.schema.typing import TStoredSchema
+
+_THREAD_POOL: ManagedThreadPool = ManagedThreadPool(1)
+TRACE_URL_SUFFIX = "/trace"
+STATE_URL_SUFFIX = "/state"
+
+class TSchemaSyncPayload(TypedDict):
+    pipeline_name: str
+    destination_name: str
+    destination_displayable_credentials: str
+    destination_fingerprint: str
+    dataset_name: str
+    schemas: List[TStoredSchema]
+
+def _send_trace_to_platform(trace: PipelineTrace, pipeline: SupportsPipeline) -> None:
+    """
+    Send the full trace after a run operation to the platform
+    TODO: Migrate this to open telemetry in the next iteration
+    """
+    if not pipeline.runtime_config.dlthub_dsn:
+        return
+
+    def _future_send() -> None:
+        try:
+            trace_dump = json.dumps(trace.asdict())
+            url = pipeline.runtime_config.dlthub_dsn + TRACE_URL_SUFFIX
+            response = requests.put(url, data=trace_dump)
+            if response.status_code != 200:
+                logger.debug(f"Failed to send trace to platform, response code: {response.status_code}")
+        except Exception as e:
+            logger.debug(f"Exception while sending trace to platform: {e}")
+
+    _THREAD_POOL.thread_pool.submit(_future_send)
+
+        # trace_dump = json.dumps(trace.asdict(), pretty=True)
+        # with open(f"trace.json", "w") as f:
+        #     f.write(trace_dump)
+
+def _sync_schemas_to_platform(trace: PipelineTrace, pipeline: SupportsPipeline) -> None:
+    if not pipeline.runtime_config.dlthub_dsn:
+        return
+
+    # sync only if load step was processed
+    load_info: LoadInfo = None
+    for step in trace.steps:
+        if step.step == "load":
+            load_info = cast(LoadInfo, step.step_info)
+
+    if not load_info:
+            return
+
+    payload = TSchemaSyncPayload(
```

Comment:
looks like PipelineStateSync. LGTM for now. we can add more data to it when we need it. we could also add the pipeline state. it will help with debugging. there's a method to retrieve it in the pipeline and methods to serialize it

Reply:
renamed it